packages/calcite-components/src/components/list/list.tsx

```diff
@@ -558,6 +558,7 @@ export class List
                         aria-label={filterPlaceholder}
                         disabled={disabled}
                         items={dataForFilter}
+                        matchFields={["label", "description", "metadata"]}
```

Comment:
I think this also needs to include `value`. Otherwise, we're changing existing filtering behavior.

Reply:
I think we should consider this a bug. It was never intended to search by value. Value is more of a unique identifier.

Reply:
For filtering, we should only be searching content. I wouldn't consider `value` to be content. I think we should not search it by default for anyone using a list with a filter.

Reply:
How about I move the list changes to a separate followup PR with the title `fix(list): only filter by list item content`? @jcfranco

Reply:
> I think we should consider this a bug. It was never intended to search by value. Value is more of a unique identifier and not content like label, description and metadata are.

I disagree with this. The filtering has always included `value` props by default. Could we introduce a `matchFields` prop to all filtering components, allowing users to omit `value` if they prefer?

Reply:
@jcfranco I reverted the list changes in this PR. Can we discuss it as part of #9569? I can create a separate issue for adding matchFields to list if we decide to do that.

 Is this PR good to merge?

Reply:
Gotcha. Will give this another look.